var/spack/repos/builtin/packages/py-pyqt5/package.py

```diff
@@ -52,7 +52,7 @@ def configure_args(self):
         args = [
             '--pyuic5-interpreter', self.spec['python'].command.path,
             '--sipdir', self.prefix.share.sip.PyQt5,
-            '--stubsdir', join_path(site_packages_dir, 'PyQt5'),
+            '--stubsdir', join_path(self.spec['python'].package.site_packages_dir, 'PyQt5'),
```

Comment:
Are you sure this works? I think `site_packages_dir` refers to `pyqt5-prefix/lib/python-x.y/site-packages` while `self.spec['python'].package.site_packages_dir` would refer to `python-prefix/lib/python-x.y/site-packages`

Reply:
It does work and I was just getting ready to ask for review. It gets the package directory. Here is the configure line showing the substitutions.
```
==> 56670: py-pyqt5: Executing phase: 'configure'
==> [2020-02-24-18:02:08.824597, 491] '/opt/packages/gpjohnsn/spack/opt/spack/linux-centos7-broadwell/gcc-9.1.0/python-3.7.4-woyhoymvmrfsdhbk5jaxijm44dhtwxqw/bin/python3.7' 'configure.py' '--sip-module=PyQt5.sip' '--bindir=/opt/packages/gpjohnsn/spack/opt/spack/linux-centos7-broadwell/gcc-9.1.0/py-pyqt5-5.13.0-d2tb7h64h4hej5xm744gxudl4pg5qwtv/bin' '--destdir=/opt/packages/gpjohnsn/spack/opt/spack/linux-centos7-broadwell/gcc-9.1.0/py-pyqt5-5.13.0-d2tb7h64h4hej5xm744gxudl4pg5qwtv/lib/python3.7/site-packages' '--incdir=/opt/packages/gpjohnsn/spack/opt/spack/linux-centos7-broadwell/gcc-9.1.0/py-pyqt5-5.13.0-d2tb7h64h4hej5xm744gxudl4pg5qwtv/include/python3.7' '--sipdir=/opt/packages/gpjohnsn/spack/opt/spack/linux-centos7-broadwell/gcc-9.1.0/py-pyqt5-5.13.0-d2tb7h64h4hej5xm744gxudl4pg5qwtv/share/sip' '--stubsdir=/opt/packages/gpjohnsn/spack/opt/spack/linux-centos7-broadwell/gcc-9.1.0/py-pyqt5-5.13.0-d2tb7h64h4hej5xm744gxudl4pg5qwtv/lib/python3.7/site-packages'
This is SIP 4.19.18 for Python 3.7.4 on linux.
The SIP code generator will be installed in
/opt/packages/gpjohnsn/spack/opt/spack/linux-centos7-broadwell/gcc-9.1.0/py-pyqt5-5.13.0-d2tb7h64h4hej5xm744gxudl4pg5qwtv/bin.
The sip.h header file will be installed in
/opt/packages/gpjohnsn/spack/opt/spack/linux-centos7-broadwell/gcc-9.1.0/py-pyqt5-5.13.0-d2tb7h64h4hej5xm744gxudl4pg5qwtv/include/python3.7.
The PyQt5.sip module will be installed in
/opt/packages/gpjohnsn/spack/opt/spack/linux-centos7-broadwell/gcc-9.1.0/py-pyqt5-5.13.0-d2tb7h64h4hej5xm744gxudl4pg5qwtv/lib/python3.7/site-packages/PyQt5.
The sip.pyi stub file will be installed in
/opt/packages/gpjohnsn/spack/opt/spack/linux-centos7-broadwell/gcc-9.1.0/py-pyqt5-5.13.0-d2tb7h64h4hej5xm744gxudl4pg5qwtv/lib/python3.7/site-packages.
The default directory to install .sip files in is
/opt/packages/gpjohnsn/spack/opt/spack/linux-centos7-broadwell/gcc-9.1.0/py-pyqt5-5.13.0-d2tb7h64h4hej5xm744gxudl4pg5qwtv/share/sip.
```
I would like to post the build log but github is rejecting the file for some reason.

Reply:
Based on the above provided output, the stub dir seems to be correct.

Reply:
the interesting thing is, when I look at the last compilation of py-pyqt5, sip.pyi is actually installed under `site_packages_dir` rather than `site_packages_dir/PyQt5`. This of course doesn't have anything to do with this PR but very strange. I will re-compile it with your latest version of pyqt5 package file. Lets see.

Reply:
It seems now, all the '*.pyi' stub files have disappeared from `site_packages_dir/PyQt5`

before:
```
sbulut@ws-067 /mnt/local/spack/opt/spack/linux-centos7-skylake/gcc-7.4.0/py-pyqt5-5.13.0-yzhwikovlapdedhcawkl3puq7i3bnzqt/lib/python3.7/site-packages/PyQt5 (try_glennpj_pyqt_etc) 
$ ls
__init__.py                    QtBluetooth.pyi  QtHelp.so                QtNetwork.pyi       QtPrintSupport.so    QtSensors.pyi     QtTest.pyi        QtX11Extras.so
__pycache__                    QtBluetooth.so   QtLocation.pyi           QtNetwork.so        QtQml.pyi            QtSensors.so      QtTest.so         QtXmlPatterns.pyi
pylupdate_main.py              QtCore.pyi       QtLocation.so            QtNfc.pyi           QtQml.so             QtSerialPort.pyi  QtWebChannel.pyi  QtXmlPatterns.so
pylupdate.so                   QtCore.so        QtMultimedia.pyi         QtNfc.so            QtQuick.pyi          QtSerialPort.so   QtWebChannel.so   QtXml.pyi
pyrcc_main.py                  QtDesigner.pyi   QtMultimedia.so          QtOpenGL.pyi        QtQuick.so           Qt.so             QtWebSockets.pyi  QtXml.so
pyrcc.so                       QtDesigner.so    QtMultimediaWidgets.pyi  QtOpenGL.so         QtQuickWidgets.pyi   QtSql.pyi         QtWebSockets.so   sip.so
_QOpenGLFunctions_2_0.so       QtGui.pyi        QtMultimediaWidgets.so   QtPositioning.pyi   QtQuickWidgets.so    QtSql.so          QtWidgets.pyi     uic
_QOpenGLFunctions_2_1.so       QtGui.so         QtNetworkAuth.pyi        QtPositioning.so    QtRemoteObjects.pyi  QtSvg.pyi         QtWidgets.so
_QOpenGLFunctions_4_1_Core.so  QtHelp.pyi       QtNetworkAuth.so         QtPrintSupport.pyi  QtRemoteObjects.so   QtSvg.so          QtX11Extras.pyi
```
after
```
sbulut@ws-067 /mnt/local/spack/opt/spack/linux-centos7-skylake/gcc-7.4.0/py-pyqt5-5.13.0-ecaafo7zt5c3dfozeekxdun7or7t2vdk/lib/python3.7/site-packages/PyQt5 (try_glennpj_pyqt_etc) 
$ ls
__init__.py        pyrcc.so                       QtCore.so      QtMultimedia.so         QtOpenGL.so        QtQuickWidgets.so   QtSql.so         QtWidgets.so      uic
__pycache__        _QOpenGLFunctions_2_0.so       QtDesigner.so  QtMultimediaWidgets.so  QtPositioning.so   QtRemoteObjects.so  QtSvg.so         QtX11Extras.so
pylupdate_main.py  _QOpenGLFunctions_2_1.so       QtGui.so       QtNetworkAuth.so        QtPrintSupport.so  QtSensors.so        QtTest.so        QtXmlPatterns.so
pylupdate.so       _QOpenGLFunctions_4_1_Core.so  QtHelp.so      QtNetwork.so            QtQml.so           QtSerialPort.so     QtWebChannel.so  QtXml.so
pyrcc_main.py      QtBluetooth.so                 QtLocation.so  QtNfc.so                QtQuick.so         Qt.so               QtWebSockets.so  sip.so
```

Reply:
> the interesting thing is, when I look at the last compilation of py-pyqt5, sip.pyi is actually installed under `site_packages_dir` rather than `site_packages_dir/PyQt5`. 

This is unrelated, and due to the way that SipPackage class is setup (see `spack edit --build-system sip`. It must have been set up so to follow sip default path conventions.

Reply:
Well, that would be because I neglected to have the prefix in the `join_path`. I will push out another commit shortly.

Reply:
Regarding the sip files, this is from the build log:
```
The PyQt5.sip module will be installed in
/opt/packages/gpjohnsn/spack/opt/spack/linux-centos7-broadwell/gcc-9.1.0/py-pyqt5-5.13.0-d2tb7h64h4hej5xm744gxudl4pg5qwtv/lib/python3.7/site-packages/PyQt5.
The sip.pyi stub file will be installed in
/opt/packages/gpjohnsn/spack/opt/spack/linux-centos7-broadwell/gcc-9.1.0/py-pyqt5-5.13.0-d2tb7h64h4hej5xm744gxudl4pg5qwtv/lib/python3.7/site-packages.
```